tests/test_config.py

```diff
@@ -30,6 +28,12 @@
     }
 )
 
+# Tests use an unused nailgun import
+# ruff: noqa: F401
+
+# Cannot use ast.literal_eval because ServerConfig isn't a basic type
+# ruff: noqa: S307
+
```

Comment:
I'm not following your unittest concern for these particular cases, but heavy mock usage was a blocker for pathlib conversion.

For the nailgun imports, that was largely to trigger processes happening when the module is evaluated.

For the ast.literal_eval that's because the ServerConfig object isn't a basic Python type so ast.literal_eval doesn't understand it.

Reply:
It does not concern me much. Refactoring of nailgun tests is unrealistic at the moment.